motor/motorid/awsec2/metadata_cmd.go

```diff
@@ -1,55 +1,91 @@
 )
 
 const (
-	identityUrl                   = "http://169.254.169.254/latest/dynamic/instance-identity/document"
-	metadataIdentityScriptWindows = "Invoke-RestMethod -TimeoutSec 1 -URI http://169.254.169.254/latest/dynamic/instance-identity/document -UseBasicParsing | ConvertTo-Json"
+	identityUrl = "http://169.254.169.254/latest/dynamic/instance-identity/document"
+	tagNameUrl  = "http://169.254.169.254/latest/meta-data/tags/instance/Name"
 )
 
-func NewCommandInstanceMetadata(provider os.OperatingSystemProvider, pf *platform.Platform) *CommandInstanceMetadata {
+func NewCommandInstanceMetadata(provider os.OperatingSystemProvider, pf *platform.Platform, config *aws.Config) *CommandInstanceMetadata {
 	return &CommandInstanceMetadata{
 		provider: provider,
 		platform: pf,
+		config:   config,
 	}
 }
 
 type CommandInstanceMetadata struct {
 	provider os.OperatingSystemProvider
 	platform *platform.Platform
+	config   *aws.Config
 }
 
 func (m *CommandInstanceMetadata) Identify() (Identity, error) {
 	instanceDocument, err := m.instanceIdentityDocument()
 	if err != nil {
 		return Identity{}, err
 	}
-
 	// parse into struct
 	doc := imds.InstanceIdentityDocument{}
 	if err := json.NewDecoder(strings.NewReader(instanceDocument)).Decode(&doc); err != nil {
 		return Identity{}, errors.Wrap(err, "failed to decode EC2 instance identity document")
 	}
+
+	name := ""
+	// Note that the tags metadata has to be enabled for this to work. If not, we fallback to trying to get the name
+	// via the aws API (if there's a config provided)
+	taggedName, err := m.instanceTags()
+	if err == nil {
+		name = taggedName
+	} else if m.config != nil {
+		ec2svc := ec2.NewFromConfig(*m.config)
+		ctx := context.Background()
+		filters := []ec2types.Filter{
+			{
+				Name:   aws.String("resource-id"),
+				Values: []string{doc.InstanceID},
+			},
+		}
+		tags, err := ec2svc.DescribeTags(ctx, &ec2.DescribeTagsInput{Filters: filters})
+		if err == nil {
+			for _, t := range tags.Tags {
+				if t.Key != nil && *t.Key == "Name" && t.Value != nil {
+					name = *t.Value
+				}
+			}
+		}
+	}
 	return Identity{
-		InstanceID: MondooInstanceID(doc.AccountID, doc.Region, doc.InstanceID),
-		AccountID:  "//platformid.api.mondoo.app/runtime/aws/accounts/" + doc.AccountID,
+		InstanceName: name,
+		InstanceID:   MondooInstanceID(doc.AccountID, doc.Region, doc.InstanceID),
+		AccountID:    "//platformid.api.mondoo.app/runtime/aws/accounts/" + doc.AccountID,
 	}, nil
 }
 
-func (m *CommandInstanceMetadata) instanceIdentityDocument() (string, error) {
+func curlWindows(url string) string {
+	return fmt.Sprintf("Invoke-RestMethod -TimeoutSec 1 -URI %s -UseBasicParsing | ConvertTo-Json", url)
+}
+
+func (m *CommandInstanceMetadata) curlDocument(url string) (string, error) {
 	switch {
 	case m.platform.IsFamily(platform.FAMILY_UNIX):
-		cmd, err := m.provider.RunCommand("curl " + identityUrl)
+		cmd, err := m.provider.RunCommand("curl " + url)
```

Comment:
Not all Linux distros have curl installed by default.

Perhaps for a follow-up PR: We could replace this with go. That should also work on both platforms.

Reply:
This is only the remove option, which is difficult to implement via go. We could only fallback to other curl-like methods.